### ISSUE TITLE
Compare streams

### ISSUE THREAD

**Issue by KleeUT** [marcdacz/compare-pdf#15]
It would be great to be able to pass pdf files in as streams rather than comparing based on files on disk.  

My use case for this is when comparing two pdfs that are stored in cloud storage. To be more specific I want to use AWS Lambda to compare two pdfs that have previously been generated and stored in AWS S3.

**Comment by marcdacz** [comment]
Interesting request. Another issue (#17) was raised asking to support buffer. Would it be sufficient to use that one too or stream is really needed as well?

**Comment by KleeUT** [comment]
Sorry for the slow reply.  For the 80% use case I think a Stream and Buffer would act similar in this case. The advantage we get when using a stream is we should be able to start generating the hash or image as the file is read into memory rather than having to read the whole thing in before starting the comparison.

**Comment by marcdacz** [comment]
I see what you are saying. I've just published the support for buffer. I'll look into this next but in the meantime, have a go with the buffer and would love to get your feedback.

Cheers

**Comment by marcdacz** [comment]
Closing this one for now since we have not received any other interests in using Stream and we already have support for Buffers.
